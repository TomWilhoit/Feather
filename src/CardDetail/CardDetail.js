import React from "react";
import "./CardDetail.scss";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";

export const CardDetail = props => {
  return (
    <div className="card-detail">
      <div className="details">
      <Link to="/">
        <button className="back-btn" onClick={props.returnToContainer}>
          BACK
        </button>
      </Link>
        <h2>{props.locations[props.singleCard].ficName}</h2>
        <h4>{props.locations[props.singleCard].realName}</h4>
        <h2>{props.locations[props.singleCard].reason}</h2>
      </div>
    </div>
  );
};

CardDetail.propTypes = {
  locations: PropTypes.array,
  weather: PropTypes.array
};

export const mapStateToProps = state => ({
  locations: state.locations,
  weather: state.weather
});

export default connect(
  mapStateToProps,
  null
)(CardDetail);
