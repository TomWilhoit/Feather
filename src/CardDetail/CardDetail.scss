.card-detail {
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.details {
  font-family: "Josefin Sans", sans-serif;
  width: 80%;
  padding: 25px;
  background: darkseagreen;
  color: white;
  text-decoration: none;
  display: flex;
  flex-direction: column;
  height: 55%;
}

.back-btn {
  margin-top: 15px;
}
