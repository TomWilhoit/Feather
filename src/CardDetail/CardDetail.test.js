import React from "react";
import { shallow } from "enzyme";
import { CardDetail } from "./CardDetail";
import { mapStateToProps } from "./CardDetail";

describe("CardDetail", () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      locations: [
        {
          realName: "Chattanooga",
          ficName: "Chattavegas",
          reason: "I know. It is so dumb"
        },
        {
          realName: "Nashville",
          ficName: "Nashvegas",
          reason: "I know. It is so dumb"
        }
      ],
      singleCard: 1
    };
    wrapper = shallow(<CardDetail {...mockProps} />);
  });

  it("should match the snapshot with all data passed in correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("mapStateToProps", () => {
  it("should return an object with the locations array", () => {
    const mockState = {
      locations: ["Chattanooga", "Nashville"]
    };
    const expected = {
      locations: ["Chattanooga", "Nashville"]
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
