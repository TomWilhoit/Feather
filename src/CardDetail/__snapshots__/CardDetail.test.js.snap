// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetail should match the snapshot with all data passed in correctly 1`] = `
<div
  className="card-detail"
>
  <div
    className="details"
  >
    <Link
      to="/"
    >
      <button
        className="back-btn"
      >
        BACK
      </button>
    </Link>
    <h2>
      Nashvegas
    </h2>
    <h4>
      Nashville
    </h4>
    <h2>
      I know. It is so dumb
    </h2>
  </div>
</div>
`;
